src/SmoFacade/Server.cs

```diff
@@ -295,6 +295,29 @@ public void AddRole(LoginProperties login, RoleProperties role)
       matchingLogin.AddRole(role.Name);
     }
 
+    public User AddUser(UserProperties user, string database)
+    {
+      var db = Databases.SingleOrDefault(d => d.Name.Equals(database, StringComparison.InvariantCultureIgnoreCase));
+      if (db == null)
+      {
+        throw new Exception("No matching database found");
+      }
+
+      return db.AddUser(user);

```

Comment:
This is fine but it could be more concise

`db == null ? throw new Exception("No matching database found"): return db.AddUser(user)`

Same for `DropUser`

Reply:
I think you got the wrong variable here.  `null_ ? ... : ...` is always going to be null

Reply:
I didn't think you could optionally return or throw with a ternary operator but apparently you can! 
https://stackoverflow.com/questions/42209135/c-sharp-operator-and-throw-exception
The same won't work for `DropUser` since it's `void`.

On further review though, I think I'll remove these methods from `Server` outright, since that seems to be more aligned with the overall design in the repo. It makes more sense either way since users are strictly database objects. 

Additionally, the method `Database()` in server https://github.com/factset/AgDatabaseMove/blob/master/src/SmoFacade/Server.cs#L163
simply returns null in the case a database can't be found. Checking if the database exists relies on this, but the other methods either mark the result of `Database()` as nullable or blindly call a method on it. 

Perhaps `Database()` should explicitly throw? Certain methods like `DropAllLogins` may fail without indication, while others may return a null reference error where a more specific exception message might be warranted. This would be enough change to warrant a new PR however.

For this specific instance I want an exception thrown if a user fails to create -- not as vital if it does so when a user fails to drop, as my changes will reflect.